src/main/java/org/sagebionetworks/bridge/sdk/models/schedules/SurveyReference.java

```diff
@@ -21,9 +27,10 @@ public static final boolean isSurveyRef(String ref) {
     
     public SurveyReference(String ref) {
         checkNotNull(ref);
-        String[] parts = ref.split(SURVEY_PATH_FRAGMENT)[1].split("/");
-        this.guid = parts[0];
-        this.createdOn = PUBLISHED_FRAGMENT.equals(parts[1]) ? null : parts[1];
+        Matcher m = p.matcher(ref);
+        m.find();
+        this.guid = m.group(1);
+        this.createdOn = (PUBLISHED_FRAGMENT.equals(m.group(2))) ? null : m.group(2);
```

Comment:
If you choose to add the tests on the server, please add the corresponding tests to the Java SDK as well. (This is another data point in favor of a shared types library.)